Adds iframe allowed URL management

valentinyanakiev opened the pull request:
Adds mutations to manage the iframe allowed URLs.
It allows to add and remove URLs from the settings
of the platform.

resolves #5014 


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Enabled platform administrators to dynamically add allowed URL entries for iframe content.
  - Provided functionality for administrators to remove existing allowed URL entries, enhancing control over iframe integration.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

valentinyanakiev committed:
Adds iframe allowed URL management

Adds mutations to manage the iframe allowed URLs.
It allows to add and remove URLs from the settings
of the platform.

hero101 commented on src/platform/platform-settings/platform.settings.service.ts:
```diff
@@ -27,4 +27,29 @@ export class PlatformSettingsService {
+    settings: IPlatformSettings,
+    iframeAllowedURL: string
+  ): string[] {
+    const currentUrls = settings.integration?.iframeAllowedUrls || [];
```
I would consider the missing urls as an exceptional case.

hero101 commented on src/platform/platform-settings/platform.settings.service.ts:
```diff
@@ -27,4 +27,29 @@ export class PlatformSettingsService {
+    settings: IPlatformSettings,
+    iframeAllowedURL: string
+  ): string[] {
+    const currentUrls = settings.integration?.iframeAllowedUrls || [];
```
I would consider the missing urls as an exceptional case.

hero101 commented on src/platform/platform/platform.resolver.mutations.ts:
```diff
@@ -70,4 +70,56 @@ export class PlatformResolverMutations {
+  })
+  async addIframeAllowedURL(
+    @CurrentUser() agentInfo: AgentInfo,
+    @Args('iframeAllowedURL') iframeAllowedURL: string
```
can the input name be simplified to `url`?

valentinyanakiev replied:
url is a bit ambiguous, maybe whitelistedURL?

hero101 commented on src/platform/platform/platform.resolver.mutations.ts:
```diff
@@ -70,4 +70,56 @@ export class PlatformResolverMutations {
+  })
+  async removeIframeAllowedURL(
+    @CurrentUser() agentInfo: AgentInfo,
+    @Args('iframeAllowedURL') iframeAllowedURL: string
```
can the input name be simplified to `url`?

valentinyanakiev replied:
url is a bit ambiguous, maybe whitelistedURL?

hero101 replied:
The name and the description should say it all. If you really want to also include information in the input parameter, I am ok.

valentinyanakiev committed:
Guards against uninitialized integration settings

Adds a check to ensure that integration settings are initialized
before attempting to access them. This prevents potential errors
and ensures the application behaves predictably.

valentinyanakiev committed:
Enhances iframe URL handling

Renames `iframeAllowedURL` to `whitelistedURL` for clarity.

Changes methods to `...OrFail` to indicate that they throw errors when settings are not properly initialized.

valentinyanakiev commented on src/platform/platform/platform.resolver.mutations.ts:
```diff
@@ -70,4 +70,56 @@ export class PlatformResolverMutations {
+        platform.settings,
+        iframeAllowedURL
+      );
+    await this.platformService.savePlatform(platform);
```
this is handled now in removeIframeAllowedURLOrFail

valentinyanakiev commented on src/platform/platform/platform.resolver.mutations.ts:
```diff
@@ -70,4 +70,56 @@ export class PlatformResolverMutations {
+        platform.settings,
+        iframeAllowedURL
+      );
+    await this.platformService.savePlatform(platform);
```
this is handled now in addIframeAllowedURLOrFail

valentinyanakiev committed:
Merge branch 'develop' into server-5014

valentinyanakiev commented on src/platform/platform/platform.resolver.mutations.ts:
```diff
@@ -70,4 +70,56 @@ export class PlatformResolverMutations {
+    await this.platformService.savePlatform(platform);
+
+    return platform.settings.integration.iframeAllowedUrls;
+  }
```
this requirement was explicitly stated in the story
